modules/internal/LocaleModelHelpGPU.chpl

```diff
@@ -71,7 +71,7 @@ module LocaleModelHelpGPU {
       return false; // need to move to different node
     } else {
       var origSubloc = chpl_task_getRequestedSubloc();
-      if (dsubloc==c_sublocid_any || dsubloc==origSubloc) {
```

Comment:
Should these checks now be for `c_sublocid_none`, since thats what `chpl_task_getRequestedSubloc` returns as the default?

Reply:
This is somewhat of a secondary change in this PR that resolves https://github.com/chapel-lang/chapel/issues/24259.

We can't really use `c_sublocid_none` here in the GPU locale. `c_sublocid_none` tells you that the destination of the `on` statement is the host. If that's different than `origSubloc`, that means we are moving the task from a device to host. So, we have to do a full-blown on with the sublocale change.